### ISSUE TITLE
lower() doesn't stop the app

### ISSUE THREAD

**Issue by galioy** [balderdashy/sails#3797]
**Sails version**: 0.11.3
**Node version**: 5.7.0
**NPM version**: 3.6.0
**Operating system**: OS X Yosemite

<hr/>

In a custom npm executable script (`node_modules/.bin/myScript`), I'm lifting Sails, doing some stuff, then calling `.lower()` but the app doesn't stop. Debugging inside `lower.js` shows that it's not even entered.

```
const Sails = require('sails')
  , rc = require('rc');

function run() {
  Sails.lift(rc('sails', {process: 'manual'}), (err, server)=> {
    if (err) {
      console.error(err);
    }

     // ... do some stuff

      Sails.lower({}, (err)=> {
        if(err){
          console.error(err);
        }
        return true;
      }); 
    });
  });
} 

run();
```

The app just continues running. No error is returned from `.lower()` and it doesn't have a callback result by default. I debugged in the `lower.js` file itself and turns out that that code isn't even reached...

I think the problem is that `lower()` returns a callback, but my script doesn't accept one, so how do I manage this case?

**Comment by sailsbot** [comment]
Hi @galioy!  It looks like you may have removed some required elements from the initial comment template, without which I can't verify that this post meets our [contribution guidelines](http://bit.ly/sails-issue-guide). To re-open this issue, please copy the template from [here](https://raw.githubusercontent.com/balderdashy/sails/master/.github/ISSUE_TEMPLATE), paste it at the beginning of your initial comment, and follow the instructions in the text. Then post a new comment (e.g. "ok, fixed!") so that I know to go back and check.

Sorry to be a hassle, but following these instructions ensures that we can help you in the best way possible and keep the Sails project running smoothly.

*If you feel this message is in error, or you want to debate the merits of my existence (sniffle), please contact inquiries@treeline.io.*

**Comment by galioy** [comment]
So, actually the app was getting lowered and all services (HTTP, socket, etc.) were shut down, but the app itself continued running. `process.exit()` was the solution. So, I first gracefully lower all services and then exit the app:

          Sails.lower({}, (err) => {
            if (err) {
              log.error('An error occurred while shutting the server down.');
              process.exit(1);
            }
            process.exit();
          });

**Comment by mikermcneil** [comment]
@galioy .lower() should definitely kill the Sails server.  That said, if you have any other stuff running in the same process, the actual process won't terminate until all the things are done.  For example, if, in your Sails app, you're doing something like `setInterval()` or if there's some other kind of unfinished business, that unfinished business would have to complete before the process exits.

This is tested in Sails core, but just to be sure, I checked it locally and it worked as expected:

```javascript
var sails = require('sails');

var myProgrammaticConfig = { /* ... */ };

sails.lift(myProgrammaticConfig, function (err, server) {
  if (err) {
    console.error('Could not lift Sails app!  Bailing now.  Error details:',err);
    return;
  }

   // * * * 
   // Don't do any stuff
   // (just in case that stuff is making your process hang)
   // * * * 

  sails.lower(function (err) {
    if (err){
      console.error('Could not lower Sails app for some reason...   You might need to type CTRL+C to kill the process.  Error details:',err);
      return;
    }

    // If we made it here, the process should gracefully exit,
    // provided there isn't any unfinished business in our custom code.

  }); //</after attempted .lower()>
});//</after attempted .lift()>
```


Let me know if I'm missing something!

**Comment by mikermcneil** [comment]
@galioy going to go ahead and close this, but again, please open a new issue if I'm missing something and there's a bug here.  Thanks!
